lib/fountain-outline-view.coffee

```diff
@@ -1,340 +1,343 @@
+
+      if scene.hasOwnProperty('children') and scene.children.length > 0
+        for child in scene.children
+          if child.type == 'synopsis'
+            formatted += '<div class="synopsis">' + child.title + '</div>'
+            formatted += '</li>'
+          else
+            formatted += '</li>'
+        formatted += @formatList('', scene.children)
+      formatted += '</li>'
+    formatted
+
+  findScenes: (text) ->
+    scenes = []
+    ref = text.split('\n')
+    scenes = @getNestedChildren([ref, 0, 0])
+    scenes = scenes[0]
+    @setEndlines(scenes, null, ref.length)
+    scenes
+
+  setEndlines: (scenes, nextParentSiblingLine, totalLineCount) ->
+    nextParentSiblingLine ||= totalLineCount
+    i = 0
+    while i < scenes.length
+      nextSiblingLine
+
+      # supports drag drop when scenes hidden
+      scenes[i].parentEndline = nextParentSiblingLine
+
+      # last element
+      if i == scenes.length - 1
+        nextSiblingLine = nextParentSiblingLine
+        scenes[i].endline = nextParentSiblingLine
+
+      # all other elements can use the next index
+      else
+        nextSiblingLine  = scenes[i+1].line
+        scenes[i].endline = scenes[i+1].line
+
+      #process children
+      if scenes[i].children
+        @setEndlines(scenes[i].children, nextSiblingLine)
+
+      i++
+
+  getNestedChildren: (scenes) ->
+    out = []
+    arr = scenes[0]
+    i = scenes[1]
+    depth = scenes[2]
+    while i < arr.length
+      section = arr[i].match(/^#+/)
+      if section
+        matched = arr[i].match(/^(#+)(.+)/)
+        if section[0].length > depth
+          nestedKids = @getNestedChildren([arr, i+1, section[0].length])
+          matchedTitleText = if matched then matched[2].trim() else ""
+          hasTitle = matched && matchedTitleText && matchedTitleText != "#"
+          currentScene =
+            line: i
+            title: if hasTitle then matched[2] else "(untitled section)"
+            type: "heading" + section[0].length
+            hasNote: false
+            children: nestedKids[0]
+            depth: depth
+          if nestedKids[1] > 0
+            i = nestedKids[1]
+          out.push currentScene
+        else
+          break
+      else if arr[i].match(/(^EXT\.)|(^INT\.)|(^\.[A-Z]+)|(^\s*=\s*.+)/)
+        currentScene =
+          line: i
+          title: arr[i]
+          type: "scene"
+          hasNote: false
+          depth: depth
+        if arr[i].match(/^\s*=\s*/)
+          synMatched = arr[i].match(/^\s*=\s*(.+)/)
+          currentScene.title = synMatched[1]
+          currentScene.type = "synopsis"
+        if arr[i].match(/\[\[[^\]]*\]\]/)
+          currentScene.hasNote = true
+        out.push currentScene
+        i++
+      else
+        i++ #if there is a blank move on
+    out = [out,i, depth]
+    out
+
+
+  clearScenes: (text) ->
+    @list.innerHTML = ""
+
+  changedPane: (pane) =>
+    @editorSubs.dispose()
+    if pane and (typeof pane.getText == 'function')
```

Comment:
I don't know if this could be another possible point of erroneous pane selection....